src/pages/build/Sets/Sets.tsx

```diff
@@ -1,57 +1,80 @@
-import React, { useEffect, useState, useContext } from 'react';
-import { Divider, Tabs, Empty } from 'antd';
-import styled from 'styled-components';
-import Menu from './Menu';
-import Weapons from './Weapons';
-import Armor from './Armor';
-import Jewelry from './Jewelry';
-import SetBar from './SetBar';
-import { BuildContext, SetTab } from '../BuildStateContext';
-import { ISet } from '../../../components/GearSlot';
-import GearCard from '../../../components/GearCard';
+import React, { useEffect, useState, useContext } from 'react'
+import { Divider, Tabs, Empty } from 'antd'
+import styled from 'styled-components'
+import Menu from './Menu'
+import Weapons from './Weapons'
+import Armor from './Armor'
+import Jewelry from './Jewelry'
+import SetBar from './SetBar'
+import { BuildContext, SetTab } from '../BuildStateContext'
+import { ISet } from '../../../components/GearSlot'
+import GearCard from '../../../components/GearCard'
 
-const { TabPane } = Tabs;
+const { TabPane } = Tabs
 
 const AbilityContainer = styled.div`
   flex: 2;
   overflow-y: auto;
   padding: 40px;
-`;
+`
 
 const Content = styled.div`
   flex: 3;
   display: flex;
-`;
+`
 
 export default ({ edit }: { edit: boolean }) => {
   // const [skills, setSkills] = useState([]);
-  const [state, dispatch] = useContext(BuildContext);
-  const [selectedSet, setSelectedSet] = useState<ISet | undefined>(undefined);
-  const [collapsed, setCollapsed] = useState(state!.selectedSet !== undefined);
+  const [state, dispatch] = useContext(BuildContext)
+  const [selectedSet, setSelectedSet] = useState<ISet | undefined>(undefined)
+  const [collapsed, setCollapsed] = useState(state!.selectedSet !== undefined)
 
   useEffect(() => {
     if (!edit) {
-      localStorage.setItem('buildState', JSON.stringify(state));
+      localStorage.setItem('buildState', JSON.stringify(state))
     }
-  }, [state]);
+  }, [state])
 
   useEffect(() => {
     if (state!.selectedSet) {
-      setSelectedSet(state!.selectedSet);
+      setSelectedSet(state!.selectedSet)
     }
-  }, [state!.selectedSet]);
+  }, [state!.selectedSet])
 
-  const { setTabKey } = state!;
+  const { setTabKey } = state!
   const handleTabChange = (key: string) => {
-    dispatch!({ type: 'SET_SET_TAB_KEY', payload: { setTabKey: key } });
-  };
+    dispatch!({ type: 'SET_SET_TAB_KEY', payload: { setTabKey: key } })
+  }
+  let stp
+  if (state) {
+    const {
+      bigPieceSelection,
+      smallPieceSelection,
+      jewelrySelection,
+      frontbarSelection,
+      backbarSelection
+    } = state
+
+    stp = bigPieceSelection
```

Comment:
rename to a more declarative variable name. make it const and remove the let declaration above.